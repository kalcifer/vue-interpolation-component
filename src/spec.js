import { shallowMount } from "@vue/test-utils";
import Interpolation from "./Index.vue";

describe("Interpolation", () => {
  it("renders content in the tag passed", () => {
    const wrapper = shallowMount(Interpolation, {
      propsData: {
        content: "test message",
        tag: "span"
      }
    });
    expect(wrapper).toMatchSnapshot();
  });

  it("renders the default slot", () => {
    const wrapper = shallowMount(Interpolation, {
      propsData: {
        content: "test message",
        tag: "span"
      },
      slots: {
        default: "submit"
      }
    });
    expect(wrapper).toMatchSnapshot();
  });

  it("replaces the {slot} pattern", () => {
    const wrapper = shallowMount(Interpolation, {
      propsData: {
        content: "test {slot} message",
        tag: "span"
      },
      slots: {
        default: "submit"
      }
    });
    expect(wrapper).toMatchSnapshot();
  });

  it("replaces custom {{slot}} pattern when regex prop is passed", () => {
    const wrapper = shallowMount(Interpolation, {
      propsData: {
        content: "test {{slot}} message",
        tag: "span",
        regex: /{{(.*?)}}/g
      },
      slots: {
        default: "double braces slot"
      }
    });
    expect(wrapper).toMatchSnapshot();
  });

  it("replaces multiple {slot} pattern", () => {
    const wrapper = shallowMount(Interpolation, {
      propsData: {
        content: "{slot} test {slot} message {slot}",
        tag: "span"
      },
      slots: {
        default: "submit"
      }
    });
    expect(wrapper).toMatchSnapshot();
  });

  it("replaces the {slot} pattern & sets classes on root element", () => {
    const wrapper = shallowMount(Interpolation, {
      propsData: {
        content: "test {slot} message",
        tag: "span",
        class: "test class"
      },
      slots: {
        default: "submit"
      }
    });
    expect(wrapper).toMatchSnapshot();
  });

  it("replaces the {slot} pattern & sets extra props on root element", () => {
    const wrapper = shallowMount(Interpolation, {
      propsData: {
        content: "test {slot} message",
        tag: "span",
        "aria-label": "testing interpolation"
      },
      slots: {
        default: "submit"
      }
    });
    expect(wrapper).toMatchSnapshot();
  });

  it("replaces any named slot", () => {
    const wrapper = shallowMount(Interpolation, {
      propsData: {
        content: "test {hello} message",
        tag: "span"
      },
      slots: {
        hello: "submit"
      }
    });
    expect(wrapper).toMatchSnapshot();
  });

  it("replaces any named & default slot", () => {
    const wrapper = shallowMount(Interpolation, {
      propsData: {
        content: "{slot} test {hello} message",
        tag: "span"
      },
      slots: {
        default: "default slot",
        hello: "submit"
      }
    });
    expect(wrapper).toMatchSnapshot();
  });

  it("replaces multiple named & default slots", () => {
    const wrapper = shallowMount(Interpolation, {
      propsData: {
        content: "{slot} test {hello} message {named}",
        tag: "span"
      },
      slots: {
        default: "default slot",
        hello: "submit",
        named: "named"
      }
    });
    expect(wrapper).toMatchSnapshot();
  });
});
