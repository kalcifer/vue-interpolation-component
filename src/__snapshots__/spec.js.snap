// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Interpolation renders content in the tag passed 1`] = `<span>test message</span>`;

exports[`Interpolation renders the default slot 1`] = `<span>test message</span>`;

exports[`Interpolation replaces any named & default slot 1`] = `<span>default slot test submit message</span>`;

exports[`Interpolation replaces any named slot 1`] = `<span>test submit message</span>`;

exports[`Interpolation replaces custom {{slot}} pattern when regex prop is passed 1`] = `<span>test double braces slot message</span>`;

exports[`Interpolation replaces multiple {slot} pattern 1`] = `<span>submit test submit message submit</span>`;

exports[`Interpolation replaces multiple named & default slots 1`] = `<span>default slot test submit message named</span>`;

exports[`Interpolation replaces the {slot} pattern & sets classes on root element 1`] = `<span class="test class">test submit message</span>`;

exports[`Interpolation replaces the {slot} pattern & sets extra props on root element 1`] = `<span aria-label="testing interpolation">test submit message</span>`;

exports[`Interpolation replaces the {slot} pattern 1`] = `<span>test submit message</span>`;
